libc/hdr/math_macros.h

```diff
@@ -15,6 +15,12 @@
 
 #else // Overlay mode
 
+// GCC will include CXX headers when __cplusplus is defined. This behavior
+// can be suppressed by defining _GLIBCXX_INCLUDE_NEXT_C_HEADERS.
+#include "src/__support/macros/properties/compiler.h"
+#ifdef LIBC_COMPILER_IS_GCC
```

Comment:
Do you mind to switch to
```
#if defined(__GNUC__) && !defined(__clang__)
```
directly, so that we don't have cross dependency between `hdr` folder and `src` folder.

Reply:
done